---
title: "Trajectory inference across conditions: differential expression and differential progression"
output:
  rmarkdown::html_document:
    highlight: pygments
    toc: true
    toc_depth: 3
    fig_width: 5
#bibliography: "`r system.file(package='bioc2020trajectories', 'vignettes', 'bibliography.bib')`"
vignette: >
  %\VignetteIndexEntry{bioc2020trajectories}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding[utf8]{inputenc}
---

# Trajectory inference across conditions: differential expression and differential progression

## Overview

### Instructor(s) name(s) and contact information

 - Kelly Street (kstreet@ds.dfci.harvard.edu)
 - Koen Van den Berge (koenvdberge@berkeley.edu)
 - Hector Roux de BÃ©zieux (hector.rouxdebezieux@berkeley.edu)
 
### Workshop Description

In single-cell RNA-sequencing (scRNA-seq), gene expression is assessed at the level of single cells. In dynamic biological systems, it may not be appropriate to assign cells to discrete groups, but rather a continuum of cell states may be observed, e.g. the differentiation of a stem cell population into mature cell types. This is often represented as a trajectory in reduced dimension.

Many methods have been suggested for trajectory inference. However, in this setting, it is often unclear how one should handle multiple biological groups or conditions, e.g. constructing and comparing the differentiation trajectory of a wild type versus a knock-out stem cell population.

In this workshop, we will explore methods for comparing multiple conditions in a trajectory inference analysis. We start by integrating datasets from multiple conditions into a single trajectory. By comparing the conditions along the trajectory's path, we can detect large-scale changes, indicative of differential progression. We also demonstrate how to detect subtler changes by finding genes that exhibit different behaviors between these conditions along a differentiation path.

[This vignette](https://bioconductor.org/packages/release/bioc/vignettes/slingshot/inst/doc/conditionsVignette.html) provides a more complete problem description and proposes a few analytical approaches, which will serve as the basis of our workshop.

### Pre-requisites


Software:

* Basic knowledge of _R_ syntax
* Familiarity with single-cell RNA-sequencing
* Familiarity with the `SingleCellExperiment` class

Background reading:

* The textbook "Orchestrating Single-Cell Analysis with Bioconductor" is a great reference for single-cell analysis using Bioconductor packages.
* The data used in this vignette was originally published in [McFaline-Figueroa, et al.](https://www.nature.com/articles/s41588-019-0489-5) "A pooled single-cell genetic screen identifies regulatory checkpoints in the continuum of the epithelial-to-mesenchymal transition."
* [Slingshot paper](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-018-4772-0)
* [tradeSeq paper](https://www.biorxiv.org/content/10.1101/623397v1)


### Workshop Participation

The workshop will start with an introduction to the problem and the dataset using presentation slides. Following this, we will have a lab session on how one may tackle the problem of handling multiple conditions in trajectory inference and in downstream analysis involving differential progression and differential expression.

### _R_ / _Bioconductor_ packages used

* The workshop will focus on Bioconductor packages [SingleCellExperiment](https://bioconductor.org/packages/release/bioc/html/SingleCellExperiment.html), [Slingshot](https://bioconductor.org/packages/release/bioc/html/slingshot.html), and [tradeSeq](https://bioconductor.org/packages/release/bioc/html/tradeSeq.html) 

### Time outline


| Activity                     | Time |
|------------------------------|------|
| Introduction                     | 15m  |
| Data Integration and Trajectory Inference   | 10m  |
| Differential Progression                    | 15m  |
| Differential Expression          | 15m  |
| Wrap-up and Conclusions          | 5m  |


### Workshop goals and objectives

Participants will learn how to reason about trajectories in single-cell RNA-seq data and how they may be used for interpretation of complex scRNA-seq datasets.


### Learning goals

* Reason about dynamic biological systems.
* Grasp the complexity of analyzing large scRNA-seq datasets with the goal of extracting relevant biological information.
* Understand the concepts of differential progression and differential expression along a trajectory path.

### Learning objectives

* Learn how to analyze single-cell RNA-seq data using Bioconductor packages.
* Import and explore large scRNA-seq datasets.
* Understand the challenges of trajectory inference.
* Compose analysis pipeline that allows interpretation of complex scRNA-seq datasets.
* Assess the added complexity of handling multiple conditions in these dynamic systems and how it influences the analysis pipeline.
* Discuss how the analysis pipeline can incorporate this change and evaluate it.

```{r}
suppressPackageStartupMessages({
  library(slingshot); library(SingleCellExperiment)
  library(RColorBrewer); library(scales)
  library(viridis)
})
```



## Primary Analysis


### Import dataset

```{r}
sce <- readRDS("../data/sce.rds")
```


### Integration

Our dataset contains cells collected from samples undergoing two different treatment conditions which were necessarily collected separately. Hence, we will start with an integration step to combine these two sets of cells, similar to batch correction. Our goal is to remove the technical effects of the different sample collections while preserving any true, biological differences between the two treatment groups.

Data integration and normalization are complex problems and there are a variety of methods addressing each. However, since neither is the main focus of this workshop, we elected to use an existing pipeline for these tasks. The full Seurat data integration workflow with SCTransform normalization is described in [this vignette](https://satijalab.org/seurat/v3.1/integration.html).

```{r integration, eval=FALSE}
########################
### Split by condition and convert to Seurat
########################
assays(sce)$logcounts <- log1p(assays(sce)$counts)
sceMock <- sce[ ,colData(sce)$pheno$treatment_id=='Mock']
sceTGFB <- sce[ ,colData(sce)$pheno$treatment_id=='TGFB']
library(Seurat)
soMock <- as.Seurat(sceMock)
soTGFB <- as.Seurat(sceTGFB)

########################
### Normalize
########################
soMock <- SCTransform(soMock, verbose = FALSE)
soTGFB <- SCTransform(soTGFB, verbose = FALSE)

########################
### Integrate
########################
dtlist <- list(Mock = soMock, TGFB = soTGFB)
intfts <- SelectIntegrationFeatures(object.list = dtlist, nfeatures = nrow(sce)) # maxes out at 4080 (why?)
dtlist <- PrepSCTIntegration(object.list = dtlist,
							 anchor.features = intfts)
anchors <- FindIntegrationAnchors(object.list = dtlist, normalization.method = "SCT",
								  anchor.features = intfts)
integrated <- IntegrateData(anchorset = anchors, normalization.method = "SCT")
integrated <- RunPCA(integrated)
integrated <- RunUMAP(integrated, dims = 1:50)
```


### Differential Topology

Once the two sets of cells have been integrated, we can visualize all the data in one reduced dimension. We also look at the repartition of cells in this space according to the treatment and spatial id.


```{r compute scores, eval = TRUE}
shuffle <- sample(ncol(sce))


layout(matrix(1:2, 1, byrow = TRUE))
par(mar = c(4.5,4,1,1))

plot(reducedDims(sce)$UMAP[shuffle, ],
  asp = 1, pch = 16, xlab = "UMAP-1", ylab = "UMAP-2",
  col = alpha(c(1:2)[factor(colData(sce)$pheno$treatment_id)][shuffle], alpha = .5))
legend("topright", pch = 16, col = 1:2, bty = "n", 
       legend = levels(factor(colData(sce)$pheno$treatment_id)))
plot(reducedDims(sce)$UMAP[shuffle, ], asp = 1, pch = 16, xlab = "UMAP-1", ylab = "UMAP-2", 
     col = alpha(c(3, 4)[factor(colData(sce)$pheno$spatial_id)][shuffle], alpha = .5))
legend("topright", pch = 16, col = 3:4, bty = "n", legend = levels(factor(colData(sce)$pheno$spatial_id)))
layout(1)
par(mar = c(5, 4, 4, 2) + .1)
```

We know from biological knowledge that the EMT development goes from the inner to the outer cells. The question is: should we fit a separate trajectory for each condition? We might expect the trajectory itself to be changed by the treatment if it is very large. Otherwise, the treatment will impact the expression profile of some genes but the overal trajectory will be preserved. To help asssess this, we deviced a spatial score. Regions with a very high score indicate places where the cell distribution according to treatment label is far from the overall distribution. Here, we see that, while there are some small regions of imbalance, the gloabl path along the development axis is well-balanced. This means that we can fit a global trajectory to the full dataset. 

```{r, eval = TRUE}
source("../R/proximity_score.R")
scores <- proximity_score(rd = reducedDims(sce)$UMAP, cl = colData(sce)$pheno$treatment_id,
                          k = 20, smooth = 40)
grad <- viridis::plasma(10, begin = 0, end = 1)
names(grad) <- levels(cut(scores$scaled_scores, breaks = 10))
plot(reducedDims(sce)$UMAP, col = grad[cut(scores$scaled_scores, breaks = 10)],
     asp = 1, pch = 16, xlab = "UMAP-1", ylab = "UMAP-2", cex =.8)
legend("topleft", legend=names(grad), col=grad, pch=16, bty='n', cex=2/3)
```

### Trajectory Inference

We perform trajectory inference to order the cells according to EMT progression. We  use `slingshot` for trajectory inference, with the cells' position (inner or outer) serving as the cluster identifier. This ensures that we will only find a single lineage while still allowing suffiecient flexibility to correctly orient the pseudotime axis.

Note that we perform trajectory inference using cells from both conditions, rather than splitting the data into two groups. This choice allows us to use the maximum amount of data in the construction of our trajectory, which should lead to more robust results than separate, potentially noisey trajectories constructed on subsets of the data. As we will see, not all cell types are necessarily present in every condition, so this approach ensures that our trajectory accounts for all cell types present in the overall data. 

```{r slingshot, eval = TRUE}
library(slingshot)
sce <- slingshot(sce, reducedDim = 'UMAP', clusterLabels = colData(sce)$pheno$spatial_id,
                 start.clus = 'inner', approx_points = 150)
```


```{r plotSlingshot, echo = FALSE, fig.width =  8}
layout(matrix(1:2, 1, byrow = TRUE))
par(mar = c(4.5,4,1,1))
plot(reducedDims(sce)$UMAP[shuffle, ], asp = 1, pch = 16, xlab = "UMAP-1", ylab = "UMAP-2",
  col = hcl.colors(100, alpha = .5)[cut(sce$slingPseudotime_1, breaks = 100)][shuffle])
lines(SlingshotDataSet(sce))
# Pseudotime densities (by spatial)
ds <- list( Inner = density(slingPseudotime(sce)[colData(sce)$pheno$spatial_id == "inner", 1]),
            Outer = density(slingPseudotime(sce)[colData(sce)$pheno$spatial_id == "outer", 1]))
xlim <- range(c(ds$Inner$x, ds$Outer$x))
ylim <- range(c(ds$Inner$y, ds$Outer$y))
plot(xlim, ylim, col = "white", xlab = "Pseudotime", ylab = "")
polygon(c(min(ds$Inner$x), ds$Inner$x, max(ds$Inner$x)), c(0, ds$Inner$y, 0),
  col = alpha(brewer.pal(4, "Set1")[3], alpha = .5))
polygon(c(min(ds$Outer$x), ds$Inner$x, max(ds$Inner$x)), c(0, ds$Outer$y, 0),
        col = alpha(brewer.pal(4, "Set1")[2], alpha = .5))
legend("topright", legend = c("Inner", "Outer"), 
       fill = alpha(brewer.pal(4, "Set1")[3:2], alpha = .5), bty = "n")
layout(1)
par(mar = c(5, 4, 4, 2) + .1)
```

### Differential progression

Now that we have ordered the cells by EMT progression, we can begin to address the main question: how is this progression affected by TGF-$\beta$ treatment? In this section, we interpret this question as a univariate analysis of the pseudotime values between the two groups.

```{r plotDensities, echo=FALSE, fig.height=4}
# Pseudotime densities (by treatment)
ds <- list(Mock = density(slingPseudotime(sce)[colData(sce)$pheno$treatment_id == "Mock", 1]),
           TGFB = density(slingPseudotime(sce)[colData(sce)$pheno$treatment_id == "TGFB", 1]))
xlim <- range(c(ds$Mock$x, ds$TGFB$x))
ylim <- range(c(ds$Mock$y, ds$TGFB$y))
plot(xlim, ylim, col = "white", xlab = "Pseudotime", ylab = "")
polygon(c(min(ds$Mock$x),ds$Mock$x,max(ds$Mock$x)),
		c(0,ds$Mock$y,0), col = rgb(0,0,0,.5))
polygon(c(min(ds$TGFB$x),ds$Mock$x,max(ds$Mock$x)),
		c(0,ds$TGFB$y,0), col = alpha(brewer.pal(4,'Set1')[1], alpha = .5))
legend("topright", legend = c("Mock", "TGFB"), 
       fill = alpha(c(1, brewer.pal(3, "Set1")[1]), alpha = .5), bty = "n")
```

The density estimates for the two groups show a trimodal distribution for the untreated cells, but a tendancy toward later pseudotime values in the TGF-$\beta$ treated cells. The difference is striking enough that a standard T-test would likely be significant, but we are we are interested more generally in differences between the two distributions, not just the difference of means (one could imagine a scenario in which the treated group tended toward the extremes, but the means were the same). Hence, we propose a Kolmogorov-Smirnov Test to assess whether the two groups of pseudotime values are derived from the same distribution.

```{r KStest, eval = FALSE}
########################
### Kolmogorov-Smirnov Test
########################
ks.test(slingPseudotime(sce)[colData(sce)$pheno$treatment_id == "Mock", 1],
        slingPseudotime(sce)[colData(sce)$pheno$treatment_id == "TGFB", 1])
```

As we might expect, this test is highly signifcant, so we can conclude that there are differences between the two distributions.

### Differential expression

We will now proceed to discover genes whose expression changes along the trajectory, or between the two lineages of each trajectory using the Bioconductor package `tradeSeq`. 
For each condition (i.e., mock and TGF-Beta), a smooth average expression profile along pseudotime will be estimated for each gene, using a negative binomial model.
Each differential expression hypothesis of interest will then be translated into testing a linear combination of the parameters of this smoothed expression estimate.

#### Select number of knots

```{r evaluateK, eval=FALSE}
library(tradeSeq)
pt1 <- colData(sce)$slingPseudotime_1
cw <- matrix(rep(1, ncol(sce)))
conditions <- factor(colData(sce)$treatment_id)
set.seed(3)
icMat <- evaluateK(counts = as.matrix(assays(sce)$counts),
              pseudotime=pt1,
              cellWeights=cw,
              conditions = conditions,
              nGenes=300,
              k=3:7)
```


#### Fit GAM

```{r, eval=FALSE}
set.seed(3)
sce <- fitGAM(counts = as.matrix(assays(sce)$counts),
              pseudotime=pt1,
              cellWeights=cw,
              conditions = conditions,
              nknots=5)
mean(rowData(sce)$tradeSeq$converged)
saveRDS(sce, file="../data/sce.rds")
```


#### Assess DE along pseudotime (or pseudospace)


Note that the axis represented by the trajectory in this dataset is actually the migration of cells from the epithelium to the mesenchyme and therefore could also be looked at as a space dimension, although it is likely to be correlated with chronological time, too.

To assess significant changes in gene expression as a function of pseudotime within each lineage, we use the `associationTest`, which tests the null hypothesis that gene expression is not a function of pseudotime, i.e., whether the estimated smoothers are significantly varying as a function of pseudotime within each lineage. The `lineages=TRUE` argument specifies that we would like the results for each lineage separately, asides from the default global test, which tests for significant associations across all lineages in the trajectory simultaneously. Further, we specify a log2 fold change cut-off to test against using the `l2fc` argument. 

Across both conditions, using a global test, we find $2205$ DE genes with respect to a fold change of 2 on a 5% FDR level. On a lineage-specific basis, there are about twice as much DE genes in the mock lineage (2120) as compared to the TGFB lineage (1136). Many of the DE genes in the TGFB condition are also DE in the Mock condition, around 80%. 

The authors of the original paper found $1105$ DE genes on a FDR level of $1e-10$ for the Mock condition.


```{r, eval=FALSE}
assocRes <- associationTest(sce, lineages = TRUE, l2fc=log2(2))
assocGenes <- rownames(assocRes)[which(p.adjust(assocRes$pvalue, "fdr") <= 0.05)]
mockGenes <-  rownames(assocRes)[which(p.adjust(assocRes$pvalue_lineage1_conditionMock, "fdr") <= 0.05)]
tgfbGenes <-  rownames(assocRes)[which(p.adjust(assocRes$pvalue_lineage1_conditionTGFB, "fdr") <= 0.05)]

length(assocGenes)
length(mockGenes)
length(tgfbGenes)

UpSetR::upset(fromList(list(global=assocGenes, mock=mockGenes, tgfb=tgfbGenes)),
              nintersects=3, order.by="degree")
```



##### Visualization of DE genes

Below we visualize and cluster the genes whose expression vary over pseudotime.
As was also observed in the original manuscript, genes are mainly upregulated at the start- or endpoints of the lineage.

```{r, eval=FALSE}
### based on mean smoother
yhatSmooth <- predictSmooth(sce, gene=mockGenes, nPoints=50)
heatSmooth <- pheatmap(t(scale(t(yhatSmooth[,1:50]))), cluster_cols = FALSE,
         show_rownames = FALSE, show_colnames=FALSE)

## the hierarchical trees constructed here, can also be used for 
## clustering of the genes according to their average expression pattern.
cl <- sort(cutree(heatSmooth$tree_row, k=6))
table(cl)
```


```{r, eval=FALSE}
### based on fitted values (plotting takes a while to run)
yhatCell <- predictCells(sce, gene=mockGenes)
yhatCellMock <- yhatCell[,conditions == "Mock"]
# order according to pseudotime
ooMock <- order(pt1[conditions == "Mock"], decreasing=FALSE)
yhatCellMock <- yhatCellMock[,ooMock]
pheatmap(t(scale(t(yhatCellMock))), cluster_cols = FALSE,
          show_rownames = FALSE, show_colnames=FALSE)
```

##### Gene set enrichment analysis on genes from the Mock condition

Gene set enrichment analysis on the DE genes within the mock condition confirms the biology on epithelial cell differentiation.

```{r, eval=FALSE}
## C5 category is according to gene ontology grouping: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4707969/pdf/nihms-743907.pdf
library(msigdbr)
library(fgsea)
geneSets <- msigdbr(species = "Mus musculus", category = "C5", subcategory = "BP")
### filter background to only include genes that we assessed.
geneSets$gene_symbol <- toupper(geneSets$gene_symbol)
geneSets <- geneSets[geneSets$gene_symbol %in% names(sce),]
m_list <- geneSets %>% split(x = .$gene_symbol, f = .$gs_name)
stats <- assocRes$waldStat_lineage1_conditionMock
names(stats) <- rownames(assocRes)
eaRes <- fgsea(pathways = m_list, stats=stats, nperm=5e4, minSize=10)
ooEA <- order(eaRes$pval, decreasing=FALSE)
kable(head(eaRes[ooEA,1:3], n=20))
```


#### Differential expression between conditions

In the following sections, we will investigate differential expression for each gene, between the two conditions, i.e., the mock and TGF-Beta conditions.

##### EDA

The paper mentions that CDH1 and CRB3 should be expressed in similar kinetics.
Note that the lower slope of CDH1 is also observed in the paper.

```{r, eval=FALSE}
plotSmoothers(sce, counts, gene="CDH1", alpha=1, border=TRUE) + ggtitle("CDH1") #+ scale_color_discrete(name = "Condition", labels = c("Mock", "TGFB"))
plotSmoothers(sce, counts, gene="CRB3", alpha=1, border=TRUE) + ggtitle("CRB3") #+ scale_color_discrete(name = "Condition", labels = c("Mock", "TGFB"))
```

They also mention that 'only cells treated with TGF-Beta and positioned at the outer extreme of the trajectory expressed robust levels of FN1 and CDH2'.

```{r, eval=FALSE}
plotSmoothers(sce, counts, gene="FN1", alpha=1, border=TRUE) + ggtitle("FN1") #+ scale_color_discrete(name = "Condition", labels = c("Mock", "TGFB"))
plotSmoothers(sce, counts, gene="CDH2", alpha=1, border=TRUE) + ggtitle("CDH2") #+ scale_color_discrete(name = "Condition", labels = c("Mock", "TGFB"))
```


##### Differential expression analysis

To test differential expression between conditions, we use the `conditionTest` function implemented in `tradeSeq`. 
This function tests the null hypothesis that genes have identical expression patterns in each condition.
We discover 2013 genes that are DE with a fold change higher than $2$ or lower than $1/2$.

```{r, eval=FALSE}
condRes <- conditionTest(sce, l2fc=log2(2))
condRes$padj <- p.adjust(condRes$pvalue, "fdr")
mean(condRes$padj <= 0.05, na.rm=TRUE)
sum(condRes$padj <= 0.05, na.rm=TRUE)
conditionGenes <- rownames(condRes)[condRes$padj <= 0.05]
conditionGenes <- conditionGenes[!is.na(conditionGenes)]
```


##### Visualize most and least significant gene

```{r, eval=FALSE}
# plot genes
oo <- order(condRes$waldStat, decreasing=TRUE)

# most significant gene
tradeSeq:::.plotSmoothers_conditions(sce, counts, gene=rownames(counts)[oo[1]], alpha=1, border=TRUE) #+ scale_color_discrete(name = "Condition", labels = c("Mock", "TGFB"))

# least significant gene
tradeSeq:::.plotSmoothers_conditions(sce, counts, gene=rownames(counts)[oo[nrow(counts)]], alpha=1, border=TRUE) #+ scale_color_discrete(name = "Condition", labels = c("Mock", "TGFB"))
```

#### Heatmaps of genes DE between conditions

Below we show heatmaps of the genes DE between conditions. The DE genes in the heatmaps are ordered according to a hierarchical clustering on the TGF-Beta condition.

```{r, eval=FALSE}
### based on mean smoother
yhatSmooth <- predictSmooth(sce, gene=conditionGenes, nPoints=50)
heatSmooth_TGF <- pheatmap(t(scale(t(yhatSmooth[,51:100]))), cluster_cols = FALSE,
         show_rownames = FALSE, show_colnames=FALSE, main="TGF-Beta", legend=FALSE,
         silent = TRUE)

matchingHeatmap_mock <- pheatmap(t(scale(t(yhatSmooth[heatSmooth_TGF$tree_row$order,1:50]))), 
                                 cluster_cols = FALSE, cluster_rows = FALSE,
                                 show_rownames = FALSE, show_colnames=FALSE, main="Mock",
                                 legend=FALSE, silent = TRUE)

grid.arrange(heatSmooth_TGF[[4]], matchingHeatmap_mock[[4]], ncol=2)
```

#### Gene set enrichment analysis

Gene set enrichment analysis on genes that are differentially expressed between conditions finds evidence for cell motility, cell junctions/adhesion and gastrulation. The original paper also focusses on the KRAS signaling pathway, which induces cell migration, amongst others.
Other related processes include morphogenesis, gastrulation and cell adhesion.

```{r, eval=FALSE}
statsCond <- condRes$waldStat
names(statsCond) <- rownames(condRes)
eaRes <- fgsea(pathways = m_list, stats=statsCond, nperm=5e4, minSize=10)
ooEA <- order(eaRes$pval, decreasing=FALSE)
kable(head(eaRes[ooEA,1:3], n=20))
```


